packages/neos-ui-editors/src/Editors/SelectBox/SelectBoxHelpers.js

```diff
@@ -15,3 +17,12 @@ export const processSelectBoxOptions = (i18nRegistry, selectBoxOptions) =>
             {label: i18nRegistry.translate(selectBoxOptions[k].label)}
         ))
         .filter(k => k);
+
+export const getOptionKey = (optionValueAccessor, option) => {
+    let key = optionValueAccessor(option);
+    if (typeof key === 'object') {
+        key = $get('__identity', key);
```

Comment:
I believe this should be handled differently, but I couldn't come up with a different approach.